oidc-service-impl/src/main/java/io/mosip/esignet/services/ConsentHelperService.java

```diff
@@ -160,6 +193,74 @@ private ConsentAction evaluateConsentAction(OIDCTransaction transaction, Consent
         } catch (JsonProcessingException e) {
             throw new EsignetException(ErrorConstants.INVALID_CLAIM);
         }
+        //compareing the new hash with the saved one
         return consentDetail.getHash().equals(hash) ? ConsentAction.NOCAPTURE : ConsentAction.CAPTURE;
     }
+
+    public boolean verifyConsentSignature(ConsentDetail consentDetail,OIDCTransaction transaction) {
+        try{
+            String jwtToken = generateSignedObject(consentDetail);
+            if (StringUtils.isEmpty(jwtToken)) {
+                return false;
+            }
+            SignedJWT signedJWT = SignedJWT.parse(jwtToken);
+            JWSHeader header = signedJWT.getHeader();
+            String thumbPrint=header.getX509CertSHA256Thumbprint().toString();
+            String idHash=getIndividualIdHash(authorizationHelperService.getIndividualId(transaction));
+            Optional<PublicKeyRegistry> publicKeyRegistryOptional=publicKeyRegistryService.
+                    findFirstByIdHashAndThumbprintAndExpiredtimesGreaterThanOrderByExpiredtimesDesc(idHash,thumbPrint);
+            if(publicKeyRegistryOptional.isPresent())
+            {
+                Certificate certificate=IdentityProviderUtil.convertToCertificate(publicKeyRegistryOptional.get().getCertificate());
+                PublicKey publicKey =certificate.getPublicKey();
+                JWSVerifier verifier = new RSASSAVerifier((RSAPublicKey) publicKey);
+                log.info("signed jwt {} ", signedJWT);
+                if(signedJWT.verify(verifier)){
+                    return true;
+                }
+            }
+            return false;
+        } catch (ParseException | JOSEException e)
+        {
+            throw new EsignetException(ErrorConstants.INVALID_CERTIFICATE);
```

Comment:
Invalid_certificate may not be the most appropriate exception here

Reply:
Invalid_token should be fine ?